File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/AcceptAllForTCFBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.serialize
import com.usercentrics.sdk.models.settings.UsercentricsConsentType
import com.usercentrics.sdk.services.tcf.TCFDecisionUILayer

internal class AcceptAllForTCFBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "acceptAllForTCF"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        val argsMap = call.arguments as Map<*, *>
        val consents = usercentrics.instance.acceptAllForTCF(
            fromLayer = TCFDecisionUILayer.valueOf(argsMap["fromLayer"] as String),
            consentType = UsercentricsConsentType.valueOf(argsMap["consentType"] as String)
        )
        result.success(consents.map { it.serialize() })
    }

}

File: ios/Classes/Bridge/AcceptAllForTCFBridge.swift
import Usercentrics

struct AcceptAllForTCFBridge : MethodBridge {

    let name: String = "acceptAllForTCF"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let argsDict = call.arguments as! Dictionary<String, Any>
        let fromLayer = TCFDecisionUILayer.initialize(from: argsDict["fromLayer"])!
        let consentType = UsercentricsConsentType.initialize(from: argsDict["consentType"])!
        let consents = usercentrics.shared.acceptAllForTCF(fromLayer: fromLayer, consentType: consentType)
        result(consents.map { $0.serialize() })
    }
}
